n2v/internals/N2V_DataGenerator.py

```diff
@@ -94,15 +98,23 @@ def load_imgs_from_directory(self, directory, filter='*.tif', dims='YX'):
         dims      : String, optional(default='YX')
                     Dimensions of the images to read. Known dimensions are: 'TZYXC'
 
+        names_back: if set to True, the function returns the names of the input files as list
+
+        to32bit: makes conversion to 32 bit optional, if set to False the original datatype is used
+
         Returns
         -------
         images : list(array(float))
                  A list of the read tif-files. The images have dimensionality 'SZYXC' or 'SYXC'
+        (optional): files, names of the input files as list
         """
 
         files = glob(join(directory, filter))
         files.sort()
-        return self.load_imgs(files, dims=dims)
+        if names_back:
+            return files, self.load_imgs(files, to32bit, dims=dims)
```

Comment:
I am not sure what the best practice would be here. Since this changes the return value of this method.

I think the best would be to refactor the code, such that we have two methods. The old one as it is and a new one which also returns the file list. Both these methods should probably wrap a new third method which takes a file-list and return the image-list.

I would also put the `files` as second return parameter, then the user always gets the image-list as first return value.